packages/pyodide-kernel/src/worker.ts

```diff
@@ -80,6 +85,28 @@ export class PyodideRemoteKernel {
       await micropip.install('${pipliteWheelUrl}', keep_going=True)
     `);
     }
+    console.debug('extraPackagesAndIndexes', extraPackagesAndIndexes);
+
+    if (extraPackagesAndIndexes.length > 0) {
+      // note that here pkg can be a package name or a wheel url
+      for (let { package: pkg, indexes } of extraPackagesAndIndexes) {
+        let installCmd: string;
+        if (indexes === null) {
+          installCmd = `import micropip\nawait micropip.install('${pkg}', keep_going=True)`;
+        } else {
+          installCmd = `import micropip\nawait micropip.install('${pkg}', index_urls=${JSON.stringify(indexes)}, keep_going=True)`;
+        }
```

Comment:
In both cases, if pkg ends with whl, prepend baseurl, 
and from testing, check the index urls do not end in `/`, otherwise it redirects and there are CORS issues.